src/test/scala/io/github/spark_redshift_community/spark/redshift/ConversionsSuite.scala

```diff
@@ -58,6 +58,20 @@ class ConversionsSuite extends FunSuite {
     assert(convertedRow == expectedRow)
   }
 
+  test("Regression test for parsing timestamptz (bug #25 in spark_redshift_community)") {
+    val rowConverter = createRowConverter(
+      StructType(Seq(StructField("timestampWithTimezone", TimestampType))))
+
+    val timestampWithTimezone = "2014-03-01 00:00:01.123 -0300"
+    val expectedTimestampWithTimezoneMillis = TestUtils.toMillis(
```

Comment:
I am confused on this test case. The test input is "2014-03-01 00:00:01.123 -0300", the expected value is 2014-02-01 00:00:01.123. How is the expected value computing its associated timezone or is it utc based. The test input and expected is different at the month component... I am confused.

Reply:
That month must have been a typo, check more recent revisions. On the other end, I think I was ignoring timezones completely in that revision..

I initially had a look at snowflake and they are using the old questionable SimpleDateFormat (for compatibility with java 6). See https://github.com/snowflakedb/spark-snowflake/blob/0503429a82f7e1a201b82beb564e056ab3d9814e/src/test/scala/net/snowflake/spark/snowflake/ConversionsSuite.scala#L42

I don't understand why they seem to disregard completely the timezone information (see tests with or without are the same). Then I noticed that for timestamptz Redshift seems to return a corresponding datetime str with the timezone always set to +00 (the string will be 'shifted' accordingly and correctly).  So they are probably ignoring that +00 but it doesn't look like proper timezone handling?

I'm now using DateTimeFormatter from java 8 and in the latest commit I think I convert from ZonedDateTime to -> timestamp correctly.

Reply:
Reading through Redshift documentation

https://docs.aws.amazon.com/redshift/latest/dg/r_Datetime_types.html#r_Datetime_types-timestamptz

```TIMESTAMPTZ values are UTC in user tables.```

How I understand it is that when one unloads the data, the text string having something like "2001-02-17 03:38:40+00" means this column is timestampz but the bare value is already in UTC. thus +00.